cmake/AwsPrebuildDependency.cmake

```
@@ -137,7 +137,10 @@ function(aws_get_variables_for_prebuild_dependency AWS_CMAKE_PREBUILD_ARGS)
                 OR var STREQUAL "CMAKE_MAKE_PROGRAM"
                 OR var MATCHES "^CMAKE_RUNTIME_OUTPUT_DIRECTORY"
                 OR var MATCHES "^CMAKE_ARCHIVE_OUTPUT_DIRECTORY"
-                OR var MATCHES "^CMAKE_LIBRARY_OUTPUT_DIRECTORY")
+                OR var MATCHES "^CMAKE_LIBRARY_OUTPUT_DIRECTORY"
+                OR var STREQUAL "CMAKE_OSX_ARCHITECTURES"
+                OR var STREQUAL "CMAKE_OSX_DEPLOYMENT_TARGET"
+                OR var STREQUAL "CMAKE_OSX_SYSROOT")
```

trivial: for better diffing in the next PR that touches this list
```suggestion
                OR var STREQUAL "CMAKE_OSX_SYSROOT"
        )
```